### DISCUSSION TITLE
Source Map: Original file name not returned; bundle name returned.

### DISCUSSION THREAD

**Discussion by stahlbauer** [swc-project/swc#7474] [Q&A] [answered]
When querying the source map via `span_to_filename` (from within my plugins` code), the bundle name is returned instead of the original file name. Is this a known issue? What might I do wrong?

**Comment by kwonoj** [comment]
Mind elaborate what `bundle name` exactly mean? Plugin's span_to_filename is proxy to the host's same call, so it depends to what filename host gets when it is transformed.

**Reply by stahlbauer** [reply]
Certainly: I am using SWC to process files that have been produced by bundlers like Webpack, RollUp, and so on. That is, these input files have already been produced from other files (for example, from TypeScript code) and have a source map included. I would like to be able to map back to the very original files the code was stemming from based on the source map information and use that information to control how my SWC plugin behaves.

**Reply by kwonoj** [reply, marked as answer]
Afaik it's expected behavior currently. It sounds like similar to the https://github.com/swc-project/swc/issues/1388, which doesn't consume the sourcemap for the source of the transform - swc's cli have explicit extend to inherit input sourcemap, but unsure if it's applied to the swc/core.

**Reply by stahlbauer** [reply]
Good reminder ;) Thanks. Some time passed by since touching that.
